src/test/java/ai/knowly/langtorch/preprocessing/splitter/text/TextSplitterTest.java

```diff
@@ -0,0 +1,228 @@
+        new CharacterTextSplitter(null, chunkSize, chunkOverlap);
+
+        // Expect IllegalArgumentException to be thrown
+    }
+
+    //TODO, this unit test will need improving. atm it only checks that length of our list of documents, it does not check the contents
+    @Test
+    public void testCharacterTextSplitter_createDocuments() {
+        // Arrange.
+        List<String> texts = Arrays.asList("foo bar", "baz");
+        CharacterTextSplitter splitter = new CharacterTextSplitter(" ", 3, 0);
+        Map<String, String> metadata = new HashMap<>();
+
+        Map<String, String> loc = new HashMap<>();
+        loc.put("from", String.valueOf(1));
+        loc.put("to", String.valueOf(1));
+
+        Optional<Metadatas> metadatas = Optional.of(new Metadatas(Arrays.asList(metadata, metadata)));
+        // Act.
+        List<DomainDocument> docs = splitter.createDocuments(texts, metadatas);
+
+        // Assert.
+        List<DomainDocument> expectedDocs = Arrays.asList(
+                new DomainDocument("foo", metadata),
+                new DomainDocument("bar", metadata),
+                new DomainDocument("baz", metadata)
+        );
+
+        assertEquals(expectedDocs.size(), docs.size());
+    }
+
+    //TODO, this unit test will need improving. atm it only checks that length of our list of documents, it does not check the contents
+    @Test
+    public void testCharacterTextSplitter_createDocumentsWithMetadata() {
+        // Arrange.
+        List<String> texts = Arrays.asList("foo bar", "baz");
+        CharacterTextSplitter splitter = new CharacterTextSplitter(" ", 3, 0);
+        List<Map<String, String>> metadataList = Arrays.asList(
+                new HashMap<String, String>() {{
+                    put("source", "1");
+                }},
+                new HashMap<String, String>() {{
+                    put("source", "2");
+                }}
+        );
+
+        Optional<Metadatas> metadatas = Optional.of(new Metadatas(metadataList));
+
+
+        // Act.
+        List<DomainDocument> docs = splitter.createDocuments(texts, metadatas);
+
+        // Assert.
+        List<DomainDocument> expectedDocs = Arrays.asList(
+                new DomainDocument("foo", new HashMap<String, String>() {{
+                    put("source", "1");
+                    put("from", String.valueOf(1));
+                    put("to", String.valueOf(1));
+                }}),
+                new DomainDocument("bar", new HashMap<String, String>() {{
+                    put("source", "1");
+                    put("from", String.valueOf(1));
+                    put("to", String.valueOf(1));
+                }}),
+                new DomainDocument("baz", new HashMap<String, String>() {{
+                    put("source", "2");
+                    put("from", String.valueOf(1));
+                    put("to", String.valueOf(1));
+                }})
+        );
+
+        assertEquals(expectedDocs.size(), docs.size());
```

Comment:
let's use assertThat from google truth library and you can find example in other tests to be consistent